nighres/brain/dots_segmentation.py

```diff
@@ -0,0 +1,720 @@
+        Input image containing the diffusion tensor coefficients in the
+        following order: volumes 0-5: D11, D22, D33, D12, D13, D23
+    mask: niimg
+        Binary brain mask image which limits computation to the defined volume.
+    wm_atlas: int, optional
+        Define which white matter atlas to use. Option 1 for 23 tracts [2]_ 
+        and option 2 for 39 tracts [1]_. (default is 1)
+    max_iter: int, optional
+        Maximum number of iterations in the conditional modes algorithm.
+        (default is 20)
+    convergence_threshold: float, optional
+        Threshold for when the iterated conditonal modes algorithm is considered
+        to have converged. Defined as the fraction of labels that change during
+        one step of the algorithm. (default is 0.002)
+    g0: float, optional
+        Parameter controlling the sharpness of the membership function. (default
+        is 1/42)
+    s_I: float, optional
+        Parameter controlling how isotropic label energies propagate to their
+        neighborhood. (default is 1/42)
+    save_data: bool, optional
+        Save output data to file. (default is False)
+    overwrite: bool, optional
+        Overwrite existing results. (default is False)
+    output_dir: str, optional
+        Path to desired output directory, will be created if it doesn't exist.
+    file_name: str, optional
+        Desired base name for output files with file extension, suffixes will 
+        be added.
+    atlas_file: str, optional
+        Path to atlas directory (default is stored in ATLAS_DIR)
+
+    Returns
+    ----------
+    dict
+        Dictionary collecting outputs under the following keys
+        (suffix of output files in brackets)
+
+        * segmentation (niimg): Hard segmentation of white matter.
+        * energy (niimg): Markov Random Field energies of labels.
+        
+    Notes
+    ----------
+    Algorithm details can be found in [1]_.
+
+    References
+    ----------
+    .. [1] Bazin, Pierre-Louis, et al. "Direct segmentation of the major white 
+       matter tracts in diffusion tensor images." Neuroimage (2011)
+       doi: https://doi.org/10.1016/j.neuroimage.2011.06.020
+    .. [2] Bazin, Pierre-Louis, et al. "Efficient MRF segmentation of DTI white 
+       matter tracts using an overlapping fiber model." Proceedings of the 
+       International Workshop on Diffusion Modelling and Fiber Cup (2009)
+    """
+    
+    
+    # For external tools: dipy
+    try:
+        from dipy.core.gradients import gradient_table
+        from dipy.align.transforms import AffineTransform3D
+        from dipy.align.imaffine import MutualInformationMetric, AffineRegistration
+    except ImportError:
+        print('Error: Dipy could not be imported, it is required'
+                +' in order to run this module. \n (aborting)')
+        return None
+    
+    
+    # In case of non-default atlas location
+    if atlas_file is not None:
+      ATLAS_DIR = atlas_file
```

Comment:
When I run the example I get an error:
```
Traceback (most recent call last):
  File "examples/example_05_dots_segmentation.py", line 48, in <module>
    file_name='DOTS_results.nii.gz')
  File "/home/julia/.local/lib/python3.6/site-packages/nighres/brain/dots_segmentation.py", line 483, in dots_segmentation
    atlas_path = os.path.join(ATLAS_DIR, 'dots_atlas')
UnboundLocalError: local variable 'ATLAS_DIR' referenced before assignment
```

Also, please do not reset the global variable ATLAS_DIR. Instead, implement something similar to the _check_atlas_dir function used in MGDM (doesn't have to be it's own function as this one is probably too specific for reuse). 
And note that atlas_file will point to a file, while ATLAS_DIR is a directory in which to find either the specified atlas_file or the default atlas.

Reply:
Also, it is really helpful to have descriptive commit messages. E.g. "include atlas_file parameter" instead of "update dots_segmentation.py", so I know which commit to look at for a given change :)

Reply:
I fixed the issue. Everything should work now. I reinstalled nighres and run the examples without any problems. Sorry about wasting your time.